qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -153,17 +153,23 @@ def _error(circuit, target=None, qubits=None):
             for key in keys:
                 target_op = target.operation_from_name(key)
                 # pylint: disable=unidiomatic-typecheck
-                if type(target_op) == type(inst.operation) and (
+                if isinstance(target_op, type(inst.operation)) and (
```

Comment:
with this change i think now you can remove the pylint disable that I put in here